pkg/monitor/worker.go

```diff
@@ -255,5 +298,23 @@ func (mon *monitor) workOne(ctx context.Context, log *logrus.Entry, doc *api.Ope
 		return
 	}
 
-	c.Monitor(ctx)
+	monitors = append(monitors, c)
+	allJobsDone := make(chan bool)
+	go execute(ctx, allJobsDone, &wg, monitors)
```

Comment:
So we essentially introduce concurent mechanism to run multiple monitors.
We might as well migrate the rest to use this pattern, because sequential run we have in the original code is error prone.